cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -1912,18 +1976,19 @@ private List<Field<?>> getApplicationPrimaryKeys(ApplicationId appId) {
     return getApplicationPrimaryKeys(appId.getNamespace(), appId.getApplication(), appId.getVersion());
   }
 
-  private List<Field<?>> getApplicationPrimaryKeys(String namespaceId, String appId, String versionId) {
+  private List<Field<?>> getLatestApplicationKeys(String namespaceId, String appName) {
     List<Field<?>> fields = new ArrayList<>();
     fields.add(Fields.stringField(StoreDefinition.AppMetadataStore.NAMESPACE_FIELD, namespaceId));
-    fields.add(Fields.stringField(StoreDefinition.AppMetadataStore.APPLICATION_FIELD, appId));
-    fields.add(Fields.stringField(StoreDefinition.AppMetadataStore.VERSION_FIELD, versionId));
+    fields.add(Fields.stringField(StoreDefinition.AppMetadataStore.APPLICATION_FIELD, appName));
+    fields.add(Fields.stringField(StoreDefinition.AppMetadataStore.LATEST_FIELD, "true"));
     return fields;
   }
 
-  private List<Field<?>> getApplicationKeys(String namespaceId, String appId) {
+  private List<Field<?>> getApplicationPrimaryKeys(String namespaceId, String appId, String versionId) {
```

Comment:
It's a bit strange that you change `getApplicationKeys` to `getApplicationPrimaryKeys`, and `getApplicationPrimaryKeys` to `getLatestApplicationKeys`. Why not just change `getApplicationKeys` to `getLatestApplicationKeys` directly

Reply:
agree^